### PR TITLE
bpo-44394: Ensure libexpat is linked against libm

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue44394

We need to link against `libm` to have `isnan`. See https://github.com/libexpat/libexpat/pull/510

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1767,7 +1767,7 @@ def detect_expat_elementtree(self):
                 ('XML_POOR_ENTROPY', '1'),
             ]
             extra_compile_args = []
-            expat_lib = []
+            expat_lib = ['m']
```
Comment (vstinner) [inline setup.py:1772]: Would you mind to add a comment mentioning "bpo-44394: " and explaining that libexpact uses isnan() of math.h?

Comment (pablogsal) [reply setup.py:1772]: Done

**Review by vstinner — APPROVED** [review]
LGTM.

File: setup.py [hunk 2]
```diff
@@ -1767,7 +1767,9 @@ def detect_expat_elementtree(self):
                 ('XML_POOR_ENTROPY', '1'),
             ]
             extra_compile_args = []
-            expat_lib = []
+            # bpo-44394: libexpact uses isnan() of math.h and needs linkage
+            # against the libm
```
Comment (vstinner) [inline setup.py:1771]: It seems like it's a recent libexpact change, in version 2.4.0: https://github.com/libexpat/libexpat/issues/511

I don't know if it's worth mentioning it.

Comment (pablogsal) [reply setup.py:1771]: Nah, I think is good since is the one we use currently in all branches

**Pull request merged**